Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -17,6 +17,39 @@ import NIO
 let crlf: StaticString = "\r\n"
 let headerSeparator: StaticString = ": "
 
+/* private but tests */ internal struct HTTPKeepAliveHeader {
```

Comment:
Interestingly, I think this may be too specific.

The `Keep-Alive` header field is a single case of a general type of HTTP header field: a list header field. This may appear repeated in the header block, and should be treated as though it were a single header field with many commas. For example:

```
HTTP/1.1 200 OK
Connection: keep-alive
Server: test
Connection: server, x-options
X-Options: cascade
```

should be treated as though there were a single `Connection` header, with the value `keep-alive, server, x-options`. We may want to, instead of this specific helper struct, provide an extension to `HTTPHeaders` that can provide an iterator of the fields in this kind of header.

We can then use this helper internally to handle the `Connection` header.